src/EnergyPlus/ConvectionCoefficients.cc

```diff
@@ -5223,22 +5221,22 @@ void DynamicIntConvSurfaceClassification(EnergyPlusData &state, int const SurfNu
     using Psychrometrics::PsyWFnTdpPb;
 
     // SUBROUTINE PARAMETER DEFINITIONS:
-    Real64 const g(9.81);                     // gravity constant (m/s**2)
-    Real64 const v(15.89e-6);                 // kinematic viscosity (m**2/s) for air at 300 K
-    Real64 const ActiveDelTempThreshold(1.5); // deg C, temperature difference for surfaces to be considered "active"
+    Real64 constexpr g(9.81);                     // gravity constant (m/s**2)
+    Real64 constexpr v(15.89e-6);                 // kinematic viscosity (m**2/s) for air at 300 K
+    Real64 constexpr ActiveDelTempThreshold(1.5); // deg C, temperature difference for surfaces to be considered "active"
 
     // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
     int ZoneNum(0);
     int PriorityEquipOn(0);
     Array1D_int HeatingPriorityStack({0, 10}, 0);
     Array1D_int CoolingPriorityStack({0, 10}, 0);
-    Array1D_int FlowRegimeStack({0, 10}, 0);
+    std::vector<InConvFlowRegime> FlowRegimeStack(10, InConvFlowRegime::Invalid);
```

Comment:
If size is known at compile time, should use `std::array` instead of `std::vector`. Where does the number 10 (vector size) come from?

Reply:
Umm... I have no idea.  It seems to be tied to the local variable ```EquipOnCount```

https://github.com/NREL/EnergyPlus/blob/42846a95e03cabcc18ef22f9778ca58ae6468128/src/EnergyPlus/ConvectionCoefficients.cc#L5299-L5314

@mjwitte @rraustad any ideas?

Reply:
Hmmm, `ZoneEquipList` is from ZoneHVAC:EquipmentList which is extensible, so I'm not sure where the 10 comes from. Oh, I see, it's counting how many pieces of equipment are active in the current timestep, so 10 is probably more than adequate, but somewhat arbitrary. Then it's looping through the active equipment to determine if heating or cooling is dominant and then select the appropriate convection algorithm. @EnergyArchmage can probably expand on this.

Reply:
Ok, if there is a chance that more than 10 would be needed, then it's best to leave it as a `std::vector`. One thing I would suggest, however, is replacing 10 with some centralized `constexpr int MaxZoneEquipmentOn` to embed this documentation into the code. Having random literal integers scattered around is not helpful and results in these types of threads.